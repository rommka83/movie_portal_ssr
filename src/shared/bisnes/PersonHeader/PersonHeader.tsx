import React from 'react';
import styles from './personHeader.module.css';

type PropsType = {
  photo: string;
  name: string;
  enName: string;
};

export const PersonHeader: React.FC<PropsType> = ({ photo, name, enName }) => {
  return (
    <div className={styles.personHeader}>
      <div className={styles.figure}>
        <img src={photo} className={styles.photo} alt="" />
      </div>
      <div className={styles.info}>
        <h2 className={styles.name}>{name}</h2>
        <span className={styles.alternate}>{enName}</span>
      </div>
    </div>
  );
};
