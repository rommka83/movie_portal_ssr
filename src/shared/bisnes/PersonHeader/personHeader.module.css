.personHeader {
    margin-bottom: 80px;
}

.figure {
    width: 150px;
    height: 200px;
    margin-bottom: 20px;
}

.photo {
    width: 100%;
    height: 100%;
    border-radius: 16px;
    object-fit: cover;
}

.info {
    margin-bottom: 20px;
}

.name {
    font-size: 56px;
    line-height: 56px;
    font-style: normal;
    font-weight: 700;
    margin-bottom: 20px;
}

.alternate {
    color: var(--gray-light);
    font-size: 15px;
    line-height: 22px;
    font-weight: 400;
    font-style: normal;
}

@media(max-width:600px) {
    .name {
        font-size: 40px;
        margin-bottom: 10px;
        line-height: 44px;
    }

    .personHeader {
        margin-bottom: 50px;
    }
}