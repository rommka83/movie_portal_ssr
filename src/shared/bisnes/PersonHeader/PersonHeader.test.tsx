import renderer from 'react-test-renderer';
import { PersonHeader } from './PersonHeader';

describe('проверка PersonHeader', () => {
  test('отрисовка PersonHeader ', () => {
    const component = renderer.create(<PersonHeader photo={''} name={''} enName={''} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
