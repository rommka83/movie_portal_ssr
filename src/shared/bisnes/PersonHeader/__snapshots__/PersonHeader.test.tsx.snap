// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка PersonHeader отрисовка PersonHeader  1`] = `
<div
  className="personHeader"
>
  <div
    className="figure"
  >
    <img
      alt=""
      className="photo"
      data-nimg="1"
      decoding="async"
      height={50}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src=""
      style={
        Object {
          "color": "transparent",
        }
      }
      width={50}
    />
  </div>
  <div
    className="info"
  >
    <h2
      className="name"
    />
    <span
      className="alternate"
    />
  </div>
</div>
`;
